packages/devtools_app/lib/src/memory/memory_screen.dart

```diff
@@ -651,3 +1099,140 @@ class MemoryBodyState extends State<MemoryBody>
+      'event': _event,
+      'vm': _vm,
+      'android': _android,
+    };
+  }
+
+  bool get hasSnapshot => _event.containsKey(snapshotDisplayName);
+  bool get hasAutoSnapshot => _event.containsKey(autoSnapshotDisplayName);
+  bool get hasMonitorStart => _event.containsKey(monitorStartDisplayName);
+  bool get hasMonitorReset => _event.containsKey(monitorResetDisplayName);
+  bool get hasExtensionEvents => _event.containsKey(extensionEventsDisplayName);
+  bool get hasManualGc => _event.containsKey(manualGCDisplayName);
+  bool get hasGc => _vm.containsKey(gcDisplayName);
+
+  int get extensionEventsLength => hasExtensionEvents
+      ? (_event[extensionEventsDisplayName] as List).length
+      : 0;
+
+  List<Map<String, Object>> get extensionEvents {
+    if (_extensionEvents.isEmpty)
+      _extensionEvents.addAll(_event[extensionEventsDisplayName]);
+    return _extensionEvents;
+  }
+
+  void _fetch() {
+    _event.clear();
+    _vm.clear();
+    _android.clear();
+
+    _fetchEventData(_event);
+    _fetchData(memoryState.vmChartController, _vm);
+    _fetchData(memoryState.androidChartController, _android);
+  }
+
+  void _fetchEventData(Map<String, Object> results) {
+    // Use the detailed extension events data stored in the memoryTimeline.
+    final eventInfo =
+        memoryState.controller.memoryTimeline.data[index].memoryEventInfo;
+
+    if (eventInfo.isEmpty) return;
+
+    if (eventInfo.isEventGC) results[manualGCDisplayName] = true;
+    if (eventInfo.isEventSnapshot) results[snapshotDisplayName] = true;
+    if (eventInfo.isEventSnapshotAuto) results[autoSnapshotDisplayName] = true;
+    if (eventInfo.isEventAllocationAccumulator) {
+      if (eventInfo.allocationAccumulator.isStart) {
+        results[monitorStartDisplayName] = true;
+      }
+      if (eventInfo.allocationAccumulator.isReset) {
+        results[monitorResetDisplayName] = true;
+      }
+    }
+
+    if (eventInfo.hasExtensionEvents) {
+      final List<Map<String, Object>> events = [];
+      for (ExtensionEvent event in eventInfo.extensionEvents.theEvents) {
+        if (event.customEventName != null) {
+          events.add(
+            {
+              'name': event.eventKind,
+              'custom': {
+                'name': event.customEventName,
+                'data': event.data,
+              },
+            },
+          );
+        } else {
+          events.add(
+            {
+              'name': event.eventKind,
+            },
+          );
+        }
+      }
+      if (events.isNotEmpty) {
+        results[extensionEventsDisplayName] = events;
+      }
+    }
+  }
+
+  void _fetchData(
+    ChartController chartController,
+    Map<String, Object> results,
+  ) {
+    for (var trace in chartController.traces) {
+      final theData = trace.data[index];
+      final yValue = theData.y;
+
+      // Convert enum'd string e.g., 'TraceName.capacity' to Map key'capacity', etc.
+      results[trace.name.split('.').last] = yValue;
```

Comment:
So that we avoid collisions with surprising rendering:
```dart
final key = trace.name.split('.').last;
assert(!results.containsKey(key));
results[key] = yValue;
```

Reply:
Good idea.